assert/assertions.go

```diff
@@ -1927,19 +1922,23 @@ func EventuallyWithT(t TestingT, condition func(collect *CollectT), waitFor time
 	for tick := ticker.C; ; {
 		select {
 		case <-timer.C:
-			collect.Copy(t)
+			for _, err := range lastFinishedTickErrs {
+				t.Errorf("%v", err)
+			}
 			return Fail(t, "Condition never satisfied", msgAndArgs...)
 		case <-tick:
 			tick = nil
-			collect.Reset()
 			go func() {
+				collect := new(CollectT)
 				condition(collect)
-				ch <- len(collect.errors) == 0
+				ch <- collect.errors
```

Comment:
Yes this way is more safer and is mandatory if `FailNow()` will use `runtime.Goexit()` as in #1481.

Reply:
Yeah, that makes sense. I've applied the suggestion. 

I'll _defer_ 🙂  implementing the test case ensuring errors are collected when the panic occurs to https://github.com/stretchr/testify/issues/1457 because that will depend on how we will handle it. As I understand, `runtime.Goexit()` in `CollectT.FailNow` will unblock using `require` assertions, but not ANY panic that could occur in conditions. To handle any panic, we could catch the panic in the goroutine running the condition, i.e.:

```go
collect := new(CollectT)
defer func() {
	if err := recover(); err != nil {
		collect.Errorf("Panic caught from condition: %v", err)
	}
	ch <- collect.errors
}()
condition(collect)
```

Reply:
@czeslavo You're right. I'll merge as is. I also notice that @marshall-lee has already implemented the defer in #1481 which is in the pipe.